In [12]:
import pandas as pd

In [13]:
path = '/home/ashish/projects/CATSCORE_ANALYSIS_Factorial_design_2wayanova/Twoway.xlsx'
data_frame = pd.read_excel(path)
data_frame

Value  prep_pro         college
0     500  three_hr        Business
1     580  three_hr        Business
2     540  three_hr     Engineering
3     460  three_hr     Engineering
4     480  three_hr  Artsandscience
5     400  three_hr  Artsandscience
6     460   One-day        Business
7     540   One-day        Business
8     560   One-day     Engineering
9     620   One-day     Engineering
10    420   One-day  Artsandscience
11    480   One-day  Artsandscience
12    560   10-Week        Business
13    600   10-Week        Business
14    600   10-Week     Engineering
15    580   10-Week     Engineering
16    480   10-Week  Artsandscience
17    410   10-Week  Artsandscience

In [14]:
import scipy
from statsmodels.formula.api import ols
import statsmodels.api as sm

In [15]:
form_show = 'Value ~ C(prep_pro) +  C(college) + C(prep_pro):C(college)'
model = ols(form_show,data_frame).fit()
model.summary()

/home/ashish/anaconda3/lib/python3.8/site-packages/scipy/stats/stats.py:1603: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=18
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                  Value   R-squared:                       0.759
Model:                            OLS   Adj. R-squared:                  0.545
Method:                 Least Squares   F-statistic:                     3.548
Date:                Sun, 26 Dec 2021   Prob (F-statistic):             0.0384
Time:                        00:30:35   Log-Likelihood:                -88.591
No. Observations:                  18   AIC:                             195.2
Df Residuals:                       9   BIC:                             203.2
Df Model:                           8                                         
Covariance Type:            nonrobust                                         
=====================================================================================================================
                                                        coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------------------------------------------
Intercept                                           445.0000     33.208     13.400      0.000     369.878     520.122
C(prep_pro)[T.One-day]                                5.0000     46.963      0.106      0.918    -101.238     111.238
C(prep_pro)[T.three_hr]                              -5.0000     46.963     -0.106      0.918    -111.238     101.238
C(college)[T.Business]                              135.0000     46.963      2.875      0.018      28.762     241.238
C(college)[T.Engineering]                           145.0000     46.963      3.088      0.013      38.762     251.238
C(prep_pro)[T.One-day]:C(college)[T.Business]       -85.0000     66.416     -1.280      0.233    -235.244      65.244
C(prep_pro)[T.three_hr]:C(college)[T.Business]      -35.0000     66.416     -0.527      0.611    -185.244     115.244
C(prep_pro)[T.One-day]:C(college)[T.Engineering]     -5.0000     66.416     -0.075      0.942    -155.244     145.244
C(prep_pro)[T.three_hr]:C(college)[T.Engineering]   -85.0000     66.416     -1.280      0.233    -235.244      65.244
==============================================================================
Omnibus:                       16.430   Durbin-Watson:                   2.984
Prob(Omnibus):                  0.000   Jarque-Bera (JB):                2.333
Skew:                           0.000   Prob(JB):                        0.311
Kurtosis:                       1.236   Cond. No.                         13.9
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [16]:
anova_table = sm.stats.anova_lm(model,typ=2)
anova_table

sum_sq   df          F    PR(>F)
C(prep_pro)              6100.0  2.0   1.382872  0.299436
C(college)              45300.0  2.0  10.269521  0.004757
C(prep_pro):C(college)  11200.0  4.0   1.269521  0.350328
Residual                19850.0  9.0        NaN       NaN

# CONCLUSION::
## Factor A = preparation program type(3 treatments).
## Factor B = college background type(3 treatments).

        Since,
            1. p-value(factor A)=0.299 > alpha=0.05 ==> accept null hypothesis.
            2. p-value(Interaction factor)=0.350 > alpha=0.05 ==>accept null hypothesis.
            3.p-value(Factor B)=0.005 < alpha=0.05 ==>reject null hypothesis(B-factor).

# We can confidently say with a significance level of 5% that only the factor B(the type of under graduate college) can significantly effect  the CAT Score.